src/test/unit/BLSRegistryCoordinatorWithIndicesUnit.t.sol

```diff
@@ -665,6 +749,104 @@ contract BLSRegistryCoordinatorWithIndicesUnit is MockAVSDeployer {
+
+        // eject
+        cheats.prank(ejector);
+        registryCoordinator.ejectOperatorFromCoordinator(defaultOperator, quorumNumbers, defaultPubKey, operatorIdsToSwap);
+        
+        // make sure the operator is deregistered
+        assertEq(
+            keccak256(abi.encode(registryCoordinator.getOperator(defaultOperator))), 
+            keccak256(abi.encode(IRegistryCoordinator.Operator({
+                operatorId: defaultOperatorId,
+                status: IRegistryCoordinator.OperatorStatus.DEREGISTERED
+            })))
+        );
+        // make sure the operator is not in any quorums
+        assertEq(registryCoordinator.getCurrentQuorumBitmapByOperatorId(defaultOperatorId), 0);
+    } 
+
+    function testEjectOperatorFromCoordinator_SubsetOfQuorums_Valid() public {
+        // register operator with default stake with 2 quorums
+        bytes memory quorumNumbers = new bytes(2);
+        quorumNumbers[0] = bytes1(defaultQuorumNumber);
+        quorumNumbers[1] = bytes1(defaultQuorumNumber + 1);
+
+        for (uint i = 0; i < quorumNumbers.length; i++) {
+            stakeRegistry.setOperatorWeight(uint8(quorumNumbers[i]), defaultOperator, defaultStake);
+        }
+
+        cheats.prank(defaultOperator);
+        registryCoordinator.registerOperatorWithCoordinator(quorumNumbers, defaultPubKey, defaultSocket);
+
+        // eject from only first quorum
+        bytes memory quorumNumbersToEject = new bytes(1);
+        quorumNumbersToEject[0] = quorumNumbers[0];
+
+        // operator is the only one registered so they are the one with the largest index
+        bytes32[] memory operatorIdsToSwap = new bytes32[](1);
+        operatorIdsToSwap[0] = defaultOperatorId;
+
+        cheats.expectEmit(true, true, true, true, address(blsPubkeyRegistry));
+        emit OperatorRemovedFromQuorums(defaultOperator, quorumNumbersToEject);
+
+        cheats.expectEmit(true, true, true, true, address(stakeRegistry));
+        emit StakeUpdate(defaultOperatorId, uint8(quorumNumbersToEject[0]), 0);
+
+        cheats.prank(ejector);
+        registryCoordinator.ejectOperatorFromCoordinator(defaultOperator, quorumNumbersToEject, defaultPubKey, operatorIdsToSwap);
+        
+        // make sure the operator is registered
+        assertEq(
+            keccak256(abi.encode(registryCoordinator.getOperator(defaultOperator))), 
+            keccak256(abi.encode(IRegistryCoordinator.Operator({
+                operatorId: defaultOperatorId,
+                status: IRegistryCoordinator.OperatorStatus.REGISTERED
+            })))
+        );
+        // make sure the operator is not in any quorums
+        assertEq(
+            registryCoordinator.getCurrentQuorumBitmapByOperatorId(defaultOperatorId), 
+            BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers) & ~BitmapUtils.orderedBytesArrayToBitmap(quorumNumbersToEject) // quorumsRegisteredFor & ~quorumsEjectedFrom
+        );
```

Comment:
wait so it's possible for an operator to be registered but not in any quorums now (as a result of ejection)
shouldn't this be checking that the operator is *not* registered?

feels like something is off here :/

Reply:
no, this only works since `quorumNumbersToEject` is a strict subset of `quorumNumbers`. if they were the same, the operator would be deregistered.

Reply:
OK gotcha, makes sense